### PR TITLE
Add upgrade documentation and suggested upgrade path

### LINKED ISSUE #344: Add how to upgrade Dnn
## Description of documentation change
It would be nice to add the procedure to follow to upgrade Dnn and the suggested upgrade path in the docs.
Explaining how to backup the database, backup the site files, then show the suggested upgrade path and the couple gotchas with some upgrades.

## Existing documentation URL (if applicable)
https://www.dnnsoftware.com/wiki/suggested-upgrade-path

## Proposed area for new documentation (if applicable)
Install/Upgrade section

### LINKED ISSUE #417: Add 'Getting Started - Setup - Upgrades' documentation
## Description of documentation change
New documentation is needed for "Upgrades" under Getting Started > Setup.

## Existing documentation URL (if applicable)
https://dnndocs.com/content/getting-started/setup/upgrades/index.html 

## Existing documentation section (if applicable)
Entire page.

## Proposed area for new documentation (if applicable)
Entire page.

## Screenshots
n/a

## Additional context
n/a

### PR DESCRIPTION
**Pull request by david-poindexter**
Resolves #417 
Resolves #344

### CODE REVIEW

File: content/getting-started/setup/upgrades/index.md [hunk 1]
```diff
@@ -8,4 +8,63 @@ dnnversion: 09.04.03
...
\ No newline at end of file
+> [!NOTE]
+> Every upgrade scenario is unique in some shape, form or fashion. This document is by no means intended to address every scenario. Some upgrades can be quite complex and it is virtually impossible to cover every scenario. This document is intended to focus only on the upgrade basics.
+
+## Step 1 - Prepare
+Before starting the upgrade process, it is recommended to have a very clear understanding of all upgrade requirements and expectations. Here are some tips to help prepare.
+
+* Know the current version of DNN Platform.
+
+* Ensure **AutoUpgrade** setting is "False". This setting can be found within **AppSettings** section of the `web.config` file in the root. Setting this to "False" will prevent the upgrade process from triggering by normal site visitors.
```
Comment (dnndev) [inline content/getting-started/setup/upgrades/index.md]: ...from being triggered by normal site visitors.

File: content/getting-started/setup/upgrades/index.md [hunk 2]
```diff
@@ -8,4 +8,63 @@ dnnversion: 09.04.03
...
+> Every upgrade scenario is unique in some shape, form or fashion. This document is by no means intended to address every scenario. Some upgrades can be quite complex and it is virtually impossible to cover every scenario. This document is intended to focus only on the upgrade basics.
+
+## Step 1 - Prepare
+Before starting the upgrade process, it is recommended to have a very clear understanding of all upgrade requirements and expectations. Here are some tips to help prepare.
+
+* Know the current version of DNN Platform.
+
+* Ensure **AutoUpgrade** setting is "False". This setting can be found within **AppSettings** section of the `web.config` file in the root. Setting this to "False" will prevent the upgrade process from triggering by normal site visitors. 
+
+* Know the desired version of DNN Platform and download the appropriate "Upgrade" ZIP package from [the official repository on GitHub](https://github.com/dnnsoftware/DNN.Platform/releases) (which can be found in the **Assets** section for the desired version release).
```
Comment (dnndev) [inline content/getting-started/setup/upgrades/index.md]: Perhaps add an example of what the filename of the upgrade zip file looks like? Not required but may be helpful.

File: content/getting-started/setup/upgrades/index.md [hunk 3]
```diff
@@ -8,4 +8,63 @@ dnnversion: 09.04.03
...
+* If possible, plan to test the upgrade in a non-production environment prior to attempting in production.
+
+## Step 2 - Take Website Offline
+When ready to begin the upgrade process, take the website offline to avoid any access or changes to the site by other users. This can be done by adding an `app_offline.htm` file to the root of the site. This is an ASP.NET feature and the HTML file can contain any HTML you see fit to communicate with site visitors about the site unavailability. 
+
+> [!NOTE]
+> This also prevents access to any other site pages. Any attempts will be redirected to the base URL and will display the HTML message contained in the `app_offline.htm` file.
+
+## Step 3 - Backup Files & Database
+It is recommended to perform this step in a "native" fashion.
```
Comment (dnndev) [inline content/getting-started/setup/upgrades/index.md]: It is unclear what `native fashion` means.

File: content/getting-started/setup/upgrades/index.md [hunk 4]
```diff
@@ -8,4 +8,63 @@ dnnversion: 09.04.03
...
+The database can be backed up using Microsoft SQL Server Management Studio (SSMS).
+
+If "native" access to the server is not possible, there other options depending on the hosting environment and available third-party extensions.  Following is a short list of alternate backup options.
+
+* Secure File Transfer Protocol (SFTP) or File Transfer Protocol (FTP) in combination with remote Microsoft SQL Server Management Studio (SSMS).
+* Control Panel (e.g., Plesk)
+* DNN Backup (3rd-party module by Evotiva)
+
+## Step 4 - Apply Upgrade Package
+Extract the desired "Upgrade" ZIP package to the root of the website. (_If the "Upgrade" ZIP file was already extracted to a temporary location, then simply copy the extracted files to the root of the website._) If prompted, chose the option to overwrite/merge any files/folders.
```
Comment (dnndev) [inline content/getting-started/setup/upgrades/index.md]: Add an `Important` note that the zip file(s) need to be unblocked before unzipping (along with instructions on how to do that).

File: content/getting-started/setup/upgrades/suggested-upgrade-path/index.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+dnnversion: 09.04.03
+---
+
+# DNN Platform Suggested Upgrade Path
+Following is the recommend upgrade path for **DNN Platform** based on experiences of many DNN Community members.
+
+> [!WARNING]
+> While the below has been tested and used successfully, there are no guarantees. Each upgrade scenario can result in unique complexities and challenges. Therefore, each upgrade should be treated with full context in mind. For common best-practices, please use [these basic upgrade steps](xref:setup-upgrades) to provide guidance.
+
+The below is, of course, subject to change. If your current version of DNN Platform is between any of the versions listed below, first upgrade to the closest listed version.
```
Comment (dnndev) [inline content/getting-started/setup/upgrades/suggested-upgrade-path/index.md]: For instance, if you are starting with version 09.02.00, you should upgrade first to version 09.03.02.

File: content/getting-started/setup/upgrades/index.md (hunk 1 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: Okay, I'll submit a new PR to change this.  This PR has already been reviewed, approved and merged by @donker.  ;-)

File: content/getting-started/setup/upgrades/index.md (hunk 2 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: I suppose we could, but the file naming convention has changed over the years.

File: content/getting-started/setup/upgrades/index.md (hunk 3 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: See the sentences following that one.  That said, if you have a recommendation for better wording, feel free to change.

File: content/getting-started/setup/upgrades/index.md (hunk 4 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: These are "basic" instructions, so wasn't really planning to go into that level of detail.  If you think it is necessary, feel free to add.  I have honestly never had to "unblock" these files, so I personally consider that more of an edge case.  I could be wrong though.

File: content/getting-started/setup/upgrades/suggested-upgrade-path/index.md (hunk 5 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/suggested-upgrade-path/index.md]: Okay, good idea.  I'll create another PR for this.

File: content/getting-started/setup/upgrades/index.md (hunk 2 above)
Comment (dnndev) [reply content/getting-started/setup/upgrades/index.md]: Yeah, I'm aware of the different conventions. Not a big deal, but I'm thinking of it from a newbie perspective. Sometimes it's better to be explicit and not assume they know what "Upgrade" file is. I think if we said: While the naming convention has changed over the years, the file will contain the word "Upgrade" in it. For instance, for version 9.4.4, the filename is  DNN_Platform_9.4.4_Upgrade.zip" . Again, just a thought.

Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: I am including this change in a new PR.

File: content/getting-started/setup/upgrades/index.md (hunk 3 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: After re-reading it fully, I agree - it is a clunky choice of words.  I will rewrite and submit in a new PR.

File: content/getting-started/setup/upgrades/index.md (hunk 1 above)
Comment (dnndev) [reply content/getting-started/setup/upgrades/index.md]: @donker was not cut out to be an editor ;)

File: content/getting-started/setup/upgrades/index.md (hunk 4 above)
Comment (david-poindexter) [reply content/getting-started/setup/upgrades/index.md]: I am adding this in a new PR, but will try to keep it as concise as possible.

**Pull request merged**